test/models/hosts/managed_test.rb

```diff
@@ -122,5 +122,31 @@ class ManagedTest < ActiveSupport::TestCase
         assert_equal 'boot/Solaris-10.8-x86.miniroot', host.miniroot
       end
     end
+
+    describe 'hooks' do
+      let(:host) { FactoryBot.create(:host, :managed) }
+      let(:callback) { -> {} }
+
+      test 'build_entered event is sent when build set to true' do
+        ActiveSupport::Notifications.subscribed(callback, 'build_entered.foreman') do
+          callback.expects(:call).with do |_name, _started, _finished, _unique_id, payload|
+            payload == { id: host.id, hostname: host.hostname }
+          end
+
+          host.update(build: true)
+        end
+      end
+
+      test 'build_exited event is sent when build set to false' do
+        ActiveSupport::Notifications.subscribed(callback, 'build_exited.foreman') do
+          callback.expects(:call).with do |_name, _started, _finished, _unique_id, payload|
+            payload == { id: host.id, hostname: host.hostname }
+          end
+
+          host.update(build: true)
+          host.update(build: false)
+        end
+      end
+    end
   end
 end
```

Comment:
Can you also add an actual AR event test? Looks like it might not get sent actually.

Reply:
I'm not sure how it is supposed to work and how to test it. What do you think about this:
```ruby
describe 'AR hooks' do
  let(:host) { FactoryBot.build(:host, :managed) }

  test 'call #build_hooks on commit' do
    host.expects(:build_hooks)

    host.run_callbacks(:commit)
  end

  describe '#build_hooks' do
    context 'when build mode is enabled' do
      it 'runs :build callbacks' do
        host.stubs(:build?).returns(true)
        host.stubs(:previous_changes).returns({ 'build' => [false, true] })

        host.expects(:run_callbacks).with(:build)

        host.build_hooks
      end
    end

    context 'when build mode is disabled' do
      it 'runs :provision callbacks' do
        host.stubs(:build?).returns(false)
        host.stubs(:previous_changes).returns({ 'build' => [true, false] })

        host.expects(:run_callbacks).with(:provision)

        host.build_hooks
      end
    end
  end
end
```
Do you think it makes any sense?

Reply:
I thought literally creating a record in the DB, this mocking is not very useful. Some Rails AR regressions hit us in the past but you know what let's just ignore this we would easily find out.